pytest_hot_reloading/plugin.py

```diff
@@ -85,6 +87,48 @@ def _jurigged_logger(x: str) -> None:
     """
 
 
+OrigFunctionDefinition = jurigged_codetools.FunctionDefinition
+
+
+class NewFunctionDefinition(OrigFunctionDefinition):
+    def reevaluate(self, new_node, glb):
+        new_node = self.apply_assertion_rewrite(new_node, glb)
+        return super().reevaluate(new_node, glb)
+
+    def apply_assertion_rewrite(self, ast_func, glb):
+        from _pytest.assertion.rewrite import AssertionRewriter
+
+        nodes: list[ast.AST] = [ast_func]
+        while nodes:
+            node = nodes.pop()
+            for name, field in ast.iter_fields(node):
+                if isinstance(field, list):
+                    new: list[ast.AST] = []
+                    for i, child in enumerate(field):
+                        if isinstance(child, ast.Assert):
+                            # Transform assert.
+                            new.extend(
+                                AssertionRewriter(glb["__file__"], None, None).visit(child)
```

Comment:
contemplating to `glb.get("__file__")` but can't think of a case where this would fail and don't want to introduce cargo code